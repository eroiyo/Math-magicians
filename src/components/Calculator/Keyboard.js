import React, { PureComponent } from 'react';

const Keyboard = class extends PureComponent {
  render() {
    return (
        <div className="options">
          <button className="option blue" type="button" value="AC"><span>AC</span></button>
          <button className="option blue" type="button" value="+/-"><span>+/-</span></button>
          <button className="option blue" type="button" value="%"><span>%</span></button>
          <button className="option blue" type="button" value="/"><span>/</span></button>

          <button type="button" className="option" value="7"><span>7</span></button>
          <button type="button" className="option" value="8"><span>8</span></button>
          <button type="button" className="option" value="9"><span>9</span></button>
          <button type="button" className="option blue" value="X"><span>X</span></button>

          <button type="button" className="option" value="4"><span>4</span></button>
          <button type="button" className="option" value="5"><span>5</span></button>
          <button type="button" className="option" value="6"><span>6</span></button>
          <button type="button" className="option blue" value="-"><span>-</span></button>

          <button type="button" className="option" value="1"><span>1</span></button>
          <button type="button" className="option" value="2"><span>2</span></button>
          <button type="button" className="option" value="3"><span>3</span></button>
          <button type="button" className="option blue" value="+"><span>+</span></button>

          <button className="option zero" type="button" value="0"><span>0</span></button>
          <button className="option" type="button" value="."><span>.</span></button>
          <button className="option blue" type="button" value="="><span>=</span></button>
        </div>
    );
  }
};

Keyboard.displayName = 'Keyboard';

export default Keyboard;