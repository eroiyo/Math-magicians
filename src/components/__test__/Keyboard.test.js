import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Keyboard from '../Calculator/Keyboard';

describe('The Keyboard is working', () => {
  it('the Keyboard renders correctly', () => {
    const tree = renderer.create(<Keyboard />);

    expect(tree).toMatchSnapshot();
  });

  it('click on a button calls the function', () => {
    let alterable = '0';
    const alterate = () => {
      alterable = '5';
    };
    const keyboard = render(<Keyboard onClick={alterate} />);
    const fivebutton = keyboard.getByTestId('5');
    fireEvent.click(fivebutton);
    expect(alterable).toBe('5');
  });
});
