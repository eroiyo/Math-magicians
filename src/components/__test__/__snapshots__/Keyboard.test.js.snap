// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Keyboard is working the Keyboard renders correctly 1`] = `
<div
  className="options"
>
  <button
    className="option blue"
    data-testid="AC"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    <span>
      AC
    </span>
  </button>
  <button
    className="option blue"
    data-testid="+/-"
    onClick={[Function]}
    type="button"
    value="+/-"
  >
    <span>
      +/-
    </span>
  </button>
  <button
    className="option blue"
    data-testid="%"
    onClick={[Function]}
    type="button"
    value="%"
  >
    <span>
      %
    </span>
  </button>
  <button
    className="option blue"
    data-testid="รท"
    onClick={[Function]}
    type="button"
    value="รท"
  >
    <span>
      รท
    </span>
  </button>
  <button
    className="option"
    data-testid="7"
    onClick={[Function]}
    type="button"
    value="7"
  >
    <span>
      7
    </span>
  </button>
  <button
    className="option"
    data-testid="8"
    onClick={[Function]}
    type="button"
    value="8"
  >
    <span>
      8
    </span>
  </button>
  <button
    className="option"
    data-testid="9"
    onClick={[Function]}
    type="button"
    value="9"
  >
    <span>
      9
    </span>
  </button>
  <button
    className="option blue"
    data-testid="x"
    onClick={[Function]}
    type="button"
    value="x"
  >
    <span>
      x
    </span>
  </button>
  <button
    className="option"
    data-testid="4"
    onClick={[Function]}
    type="button"
    value="4"
  >
    <span>
      4
    </span>
  </button>
  <button
    className="option"
    data-testid="5"
    onClick={[Function]}
    type="button"
    value="5"
  >
    <span>
      5
    </span>
  </button>
  <button
    className="option"
    data-testid="6"
    onClick={[Function]}
    type="button"
    value="6"
  >
    <span>
      6
    </span>
  </button>
  <button
    className="option blue"
    data-testid="-"
    onClick={[Function]}
    type="button"
    value="-"
  >
    <span>
      -
    </span>
  </button>
  <button
    className="option"
    data-testid="1"
    onClick={[Function]}
    type="button"
    value="1"
  >
    <span>
      1
    </span>
  </button>
  <button
    className="option"
    data-testid="2"
    onClick={[Function]}
    type="button"
    value="2"
  >
    <span>
      2
    </span>
  </button>
  <button
    className="option"
    data-testid="3"
    onClick={[Function]}
    type="button"
    value="3"
  >
    <span>
      3
    </span>
  </button>
  <button
    className="option blue"
    data-testid="+"
    onClick={[Function]}
    type="button"
    value="+"
  >
    <span>
      +
    </span>
  </button>
  <button
    className="option zero"
    data-testid="0"
    onClick={[Function]}
    type="button"
    value="0"
  >
    <span>
      0
    </span>
  </button>
  <button
    className="option"
    data-testid="."
    onClick={[Function]}
    type="button"
    value="."
  >
    <span>
      .
    </span>
  </button>
  <button
    className="option blue"
    data-testid="="
    onClick={[Function]}
    type="button"
    value="="
  >
    <span>
      =
    </span>
  </button>
</div>
`;
